client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyClient.java

```diff
@@ -379,6 +385,85 @@ public void onComplete() {
                     emitError(e);
                 }
             }
+            
+            @Override
+            public void operationComplete(Future<Void> future) throws Exception {
+                if (!future.isSuccess()) {
+                    subscription.cancel();
+                    emitError(future.cause());
+                } else {
+                    writeComplete();
+                }
+            }
+            
+            private void writing() {
+                while (true) {
+                    int s = writing.get();
+                    if (s == WRITE_COMPLETED_NOT_WRITABLE) {
+                        if (writing.compareAndSet(s, WRITING_NOT_WRITABLE)) {
+                            break;
+                        }
+                    } else if (s == WRITE_COMPLETED_WRITABLE) {
+                        if (writing.compareAndSet(s, WRITING_WRITABLE)) {
+                            break;
+                        }
+                    } else {
+                        break;
+                    }
+                }
+            }
+            
+            private void writeComplete() {
+                while (true) {
+                    int s = writing.get();
+                    if (s == WRITING_NOT_WRITABLE) {
+                        if (writing.compareAndSet(s, WRITE_COMPLETED_NOT_WRITABLE)) {
+                            break;
+                        }
+                    } else if (s == WRITING_WRITABLE) {
+                        if (writing.compareAndSet(s, WRITE_COMPLETED_WRITABLE)) {
+                            subscription.request(1);
+                            break;
+                        }
+                    } else {
+                        break;
+                    }
+                }
+            }
+
+            void channelWritable(boolean writable) {
+                while (true) {
+                    int s = writing.get();
+                    if (writable) {
+                        if (s == WRITE_COMPLETED_NOT_WRITABLE) {
+                            if (writing.compareAndSet(s, WRITE_COMPLETED_WRITABLE)) {
+                                subscription.request(1);
+                                break;
+                            }
+                        } else if (s == WRITING_NOT_WRITABLE) {
+                            if (writing.compareAndSet(s, WRITING_WRITABLE)) {
+                                break;
+                            }
+                        } else {
+                            break;
+                        }
+                    } else {
+                        if (s == WRITE_COMPLETED_WRITABLE) {
+                            if (writing.compareAndSet(s, WRITE_COMPLETED_NOT_WRITABLE)) {
+                                subscription.request(1);
```

Comment:
We reach this line if the channel transitioned from writable to not writable and request content is not currently being written to the channel, right? So does it make sense to request more in that case?

Reply:
Oops, I apparently was commenting on an outdated commit anyway.

Reply:
well spotted, you were right and yep already fixed in 67fcb17